File: src/lib/xp.ts
export type Difficulty = "easy" | "medium" | "hard" | "epic";

export const XP_REWARDS: Record<Difficulty, number> = {
  easy: 10,
  medium: 25,
  hard: 50,
  epic: 100,
};

export const DIFFICULTY_COLORS: Record<Difficulty, string> = {
  easy: "bg-emerald-100 text-emerald-700 border-emerald-300",
  medium: "bg-yellow-100 text-yellow-700 border-yellow-300",
  hard: "bg-orange-100 text-orange-700 border-orange-300",
  epic: "bg-red-100 text-red-700 border-red-300",
};

export function getLevel(xp: number): number {
  return Math.floor(Math.sqrt(xp / 50)) + 1;
}

export function getXpForLevel(level: number): number {
  return (level - 1) * (level - 1) * 50;
}

export function getLevelProgress(xp: number): {
  level: number;
  currentLevelXp: number;
  nextLevelXp: number;
  progress: number;
} {
  const level = getLevel(xp);
  const currentLevelXp = getXpForLevel(level);
  const nextLevelXp = getXpForLevel(level + 1);
  const progress = (xp - currentLevelXp) / (nextLevelXp - currentLevelXp);
  return { level, currentLevelXp, nextLevelXp, progress };
}

File: ios/TaskQuest/Models/XP.swift
import SwiftUI

enum XP {
    static let rewards: [Difficulty: Int] = [
        .easy: 10,
        .medium: 25,
        .hard: 50,
        .epic: 100,
    ]

    static let timeEstimates: [Difficulty: String] = [
        .easy: "~30 min",
        .medium: "1-3 hrs",
        .hard: "4-8 hrs",
        .epic: "Multi-day",
    ]

    static func color(for difficulty: Difficulty) -> Color {
        switch difficulty {
        case .easy: .easyGreen
        case .medium: .mediumYellow
        case .hard: .hardOrange
        case .epic: .epicRed
        }
    }

    static func getLevel(xp: Int) -> Int {
        Int(floor(sqrt(Double(xp) / 50.0))) + 1
    }

    static func getXpForLevel(_ level: Int) -> Int {
        (level - 1) * (level - 1) * 50
    }

    static func getLevelProgress(xp: Int) -> (level: Int, currentLevelXp: Int, nextLevelXp: Int, progress: Double) {
        let level = getLevel(xp: xp)
        let currentLevelXp = getXpForLevel(level)
        let nextLevelXp = getXpForLevel(level + 1)
        let progress = Double(xp - currentLevelXp) / Double(nextLevelXp - currentLevelXp)
        return (level, currentLevelXp, nextLevelXp, progress)
    }
}
